federation-js/src/service/printSupergraphSdl.ts

```diff
@@ -504,72 +492,71 @@ function printInputValue(arg: GraphQLInputField) {
   if (defaultAST) {
     argDecl += ` = ${print(defaultAST)}`;
   }
-  return argDecl;
+  return argDecl + printDeprecated(arg.deprecationReason);
 }
 
-function printDirective(directive: GraphQLDirective, options?: Options) {
+function printDirective(directive: GraphQLDirective) {
   return (
-    printDescription(options, directive) +
+    printDescription(directive) +
     'directive @' +
     directive.name +
-    printArgs(options, directive.args) +
+    printArgs(directive.args) +
     (directive.isRepeatable ? ' repeatable' : '') +
     ' on ' +
     directive.locations.join(' | ')
   );
 }
 
-function printDeprecated(
-  fieldOrEnumVal: GraphQLField<any, any> | GraphQLEnumValue,
-) {
-  if (!fieldOrEnumVal.isDeprecated) {
+function printDeprecated(reason: Maybe<string>): string {
+  if (reason == null) {
     return '';
   }
-  const reason = fieldOrEnumVal.deprecationReason;
-  const reasonAST = astFromValue(reason, GraphQLString);
-  if (reasonAST && reason !== DEFAULT_DEPRECATION_REASON) {
-    return ' @deprecated(reason: ' + print(reasonAST) + ')';
+  if (reason !== DEFAULT_DEPRECATION_REASON) {
+    const astValue = print({ kind: 'StringValue', value: reason });
+    return ` @deprecated(reason: ${astValue})`;
   }
   return ' @deprecated';
 }
 
-function printDescription<T extends { description?: Maybe<string> }>(
-  options: Options | undefined,
-  def: T,
-  indentation = '',
-  firstInBlock = true,
+// Apollo addition: support both specifiedByUrl and specifiedByURL - these
+// happen across v15 and v16.
```

Comment:
@IvanGoncharov do you intend to preserve this breaking change in v16?
https://github.com/graphql/graphql-js/issues/3156